test/jdk/com/sun/net/httpserver/HttpsParametersClientAuthTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.net.InetSocketAddress;
+
+import javax.net.ssl.SSLParameters;
+
+import com.sun.net.httpserver.HttpsConfigurator;
+import com.sun.net.httpserver.HttpsParameters;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/*
+ * @test
+ * @bug 8326381
+ * @summary verifies that the setNeedClientAuth() and setWantClientAuth()
+ *          methods on HttpsParameters class work as expected
+ * @run junit HttpsParametersClientAuthTest
+ */
+public class HttpsParametersClientAuthTest {
+
+    /**
+     * verifies that {@link HttpsParameters#setNeedClientAuth(boolean)}
+     * and {@link HttpsParameters#setWantClientAuth(boolean)} methods work
+     * as expected
+     */
+    @Test
+    public void testClientAuth() throws Exception {
```

Comment:
Just a suggestion.

Now that this test uses JUnit, why doesn't it define multiple test methods for the different HttpsParameters instances?
A single test method just focus on only one HttpsParameters instance (or test case).

If the checking on a HttpsParameters instance (or test case) fails, the remaining test cases still can be executed.

Reply:
Hello John, I've updated the PR to split these into separate test methods.